src/test/article.spec.js

```diff
@@ -0,0 +1,132 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../index';
+import BaseRepository from '../repository/base.repository';
+import { createUser, createArticle, getArticle } from './utils/helpers';
+import db from '../database/models';
+import helper from '../helpers/utils';
+
+const ARTICLES_API = '/api/v1/articles';
+
+chai.use(chaiHttp);
+
+const server = () => chai.request(app);
+
+describe('PATCH api/v1/articles/bookmark', () => {
+  beforeEach(async () => {
+    await db.Bookmark.destroy({ cascade: true, truncate: true });
+    await db.User.destroy({ cascade: true, truncate: true });
+  });
+
+  it('should bookmark an article', async () => {
+    const firstUser = await createUser();
+    const secondUser = await createUser();
+    const theArticle = await createArticle(
+      await getArticle({ authorId: secondUser.id })
+    );
+    const numberOfArticles = await BaseRepository.findAll(db.Article);
+    const numberOfBookmarks = await BaseRepository.findAll(db.Bookmark, {
+      userId: firstUser.id
+    });
+    const token = helper.jwtSigner(firstUser);
+    expect(numberOfArticles.length).to.equal(1);
+    expect(numberOfBookmarks.length).to.equal(0);
+
+    const res = await server()
+      .patch(`${ARTICLES_API}/bookmark`)
+      .set('token', token)
+      .send({ articleId: theArticle.id });
+    expect(res.status).to.equal(200);
+    expect(res.body.message).to.equal('Article Bookmarked successfully');
+
+    const newNumberOfBookmarks = await BaseRepository.findAll(db.Bookmark, {
+      userId: firstUser.id
+    });
+    expect(newNumberOfBookmarks.length).to.equal(1);
```

Comment:
Assert that the article id sent matches what was inserted into the db

Reply:
Change implemented, thank you